### PR TITLE
Clean up utf8 parameter usage

### LINKED ISSUE #6163: quanda table lngfile key too long
#### Description
for MySQL in utf8mb4, the lngfile column is set to 255 which causes issues on saving due to the way utf8mb4 requires more storage than utf8 or latin
### Steps to reproduce
1. use a MySQL utf8mb4 db
2. use the quanda feature

### Environment (complete as necessary)
- Version/Git revision:
- Database Type: MySQL
- Database Version: > 5.7 
- PHP Version:

### Additional information/references

Not every install can use inno to enable innodb_large_prefix so the only feasible solution is to limit the lngfile column to 250 length

Ref: https://www.simplemachines.org/community/index.php?topic=573832.msg4061249#msg4061249

### PR DESCRIPTION
**Pull request by sbulen**
Fixes #6163 
Closes #6494

The upgrader & installer already had mechanisms to set proper collation (utf8_general_ci), but they weren't working under certain circumstances.  Tidying up this existing logic is cleaner than trying to implement partial utf8mb4 support.  Utf8mb4 support should wait for #6409 .

**_Upgrader:_**
If $db_character_set were not utf8, and, you were upgrading into a DB with a utf8mb4 default charset/collation, you would get key too long errors as reported in #6163 .  Note for this to happen you must not be utf8 already, and you must not migrate to a new Settings file (because doing so sets $db_character_set to utf8....).  

This change prevents you from falling back to the DB default, which might use utf8mb4 (for new tables only, like qanda).

**_Installer:_**
utf8_support is now a function, no longer a variable, so any empty check was erroneous (function exists vs what the function returned...).  Fixing them still yielded complex if statements that were trying to determine if we were using utf8.  

But... we're using utf8 in 2.1...  So the if statements could go.

**_The bottom line is that in 2.1, the installer and the upgrader should NEVER fall back to using the db-level default collation/charset._**

### CODE REVIEW

**Comment by sbulen** [comment]
Testing completed.  Installs & upgrades tested in mysql 8.  Also did installs & upgrades in Windows as well as linux, under circumstances that produced the initial bug report.  All OK.

**Review by Sesquipedalian — APPROVED** [review]
